pkg/controllers/execution/execution_controller.go

```diff
@@ -86,6 +84,11 @@ func (c *Controller) Reconcile(ctx context.Context, req controllerruntime.Reques
 		return c.removeFinalizer(work)
 	}
 
+	if !isClusterReady {
+		klog.Errorf("Stop sync work(%s/%s) for cluster(%s) as cluster not ready.", work.Namespace, work.Name, cluster.Name)
+		return controllerruntime.Result{Requeue: true}, fmt.Errorf("cluster(%s) not ready", cluster.Name)
```

Comment:
Endless re-queue in case of the cluster not ready?

Reply:
> 

Need to work with failover scheduler. For execution controller, it will always re-queue in case of the cluster not ready. But when the cluster turns into not ready status, scheduler will reschedule and remove the cluster from schedule result.